snippets/base/forms.py

```diff
@@ -350,6 +350,13 @@ def clean(self):
              cleaned_data.get('client_option_addon_check_type') != 'any')):
             raise forms.ValidationError('Type add-on name to check or remove add-on check.')
 
+        # Check publish permissions per channel.
+        for channel in CHANNELS:
```

Comment:
Should this check only apply when `published` is `True`, or are these permissions designed more to prevent people from editing any snippet on a channel they don't have the publish permission for?